content/en/Tips/Differences-between-list-methods-append-and-extend.md

```diff
@@ -0,0 +1,153 @@
+
+
+
+### Python List
+A list is made by enclosing elements in square brackets [] and separating them with commas. Lists are useful for storing multiple elements in a single variable.
+
+One of Python's four inbuilt data types for storing data collections is the list. Tuple, set, and dictionary are the other three; each has a distinct purpose.
+
+**Example**:
+```python
+#an empty list
+test_list = []
+
+# a list of integers
+test_list = [2, 4, 6, 8, 10]
+
+# a list with mixed data types
+test_list = [10, "Amazing", 1.42, 'Morning']
+```
+
+### Append()
+The append method `.append()` adds an item (x) or a single element to the end of the list. This is equivalent to `a[len(a):] = [x]`
+
+**Syntax**: `<list>.append(<item>)`
+The initial list isn't changed when you use `.append()`, The method modifies the original list in memory rather than making a copy of it.
+
+**Example**:
+```python
+# Define the list
+letters = ['rock','paper']
+print(letters)
+
+# Add the string scissors to the end of the existing list
+letters.append('scissors')
+print(letters)
+```
+
+**Output**:
+```python
+['rock', 'paper']
+['rock', 'paper', 'scissors']
+```
+We can append an integer to list of integers by just calling the append method followed by the integer you plan to add to the list.
+
+**Example**:
+```python
+
+# Define the list
+x = [2, 4, 6]
+print(x)
+
+x.append(8)
+print(x)
+```
+
+**Output**:
+```python
+[2, 4, 6]
+[2, 4, 6, 8]
+```
+With the `.append()` we can also add a sequence of item to a list. The complete sequence will be added as one item to the current list if the item is a sequence, such as a list, dictionary, or tuple.
+
+**Example**:
+```python
+# list definition
+group = ["Lion", "Male", 25]
+print(group)
+
+# Add the sequnce to the existing list
+group.append(("Tiger", "Female", 17))
+print(group)
+```
+**Output**:
+```python
+["Lion", "Male", 25]
+["Lion", "Male", 25, ("Tiger", "Female", 17)]
+```
+
+
+### Extend()
+The extend method `.extend()` adds to the end of a list by appending all the items from the iterable. It allows you to join two lists together. This method is equivalent to `a[len(a):] = iterable`
+
+**Syntax**: `<list>.extend(<iterable>)`
+
+**Example**:
+```python
+#define the first list to be modified
+a = ['run', 'walk']
+print(a)
+#second list to extend
+a.extend(['jog', 'fly'])
+print(a)
+```
+
+**Output**:
+```python
+['run', 'walk']
+['run', 'walk', 'jog', 'fly']
+```
+We should be able to clearly see how the extend() works on iterable from the example above, now let us extend a list of integers for better understanding.
+
+```python
+#define the first list to be modified
+x = [1, 2, 3, 4, 5]
+
+#the second list to extend
+y = [6, 7, 8, 9, 10]
+
+#let us extend list(y) on list(x)
+x.extend(y)
+
+print(x)
+```
+
+**Output**:
+```python
+#result after printing x
+[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+```
+Note: after we extend the list y, only the list `x` is modified while the list `y` stays intact.
+
+

```

Comment:
You can add performance differences between append vs extend methods and also time complexities if possible. 

look at the below link for referenc

https://stackoverflow.com/questions/252703/what-is-the-difference-between-pythons-list-methods-append-and-extend

Reply:
OK, thanks @arungudelli 
I have added the section differentiaiting between the time complexities and perfromace.

Reply:
@zenUnicorn Did you commit the changes because I am unable to see the chagnes

Reply:
Sorry for pushing it to the main branch.